lib/instrumentation/span.js

```diff
@@ -191,6 +206,39 @@ Span.prototype._encode = function (cb) {
 
   if (!this.ended) return cb(new Error('cannot encode un-ended span'))
 
+  const payload = {
```

Comment:
REVIEW NOTE: Moved `const payload = ...` to the synchronous part of this method to gather all these fields before some refs (e.g. to `this._timer`, which is used for `self.traceId` here) are removed in `Span#end()`.